### ISSUE TITLE
Negative faithfulness result

### ISSUE THREAD

**Issue by PseudoKush** [explodinggradients/ragas#79]
I got the result for faithfulness as -1.5 for one of the rows of my dataset even though its defined scale is 0 - 1. I am currently using ragas version 0.0.9.  
This did not happen when I previously ran the version 0.0.7 on the same dataset. 

![image](https://github.com/explodinggradients/ragas/assets/56498312/819e63fc-6ca8-48de-9bf3-e6213baaf8f3)

Following is the piece of code I used : 
```
gpt3 = ChatOpenAI()
faithfulness_gpt3 = Faithfulness(
    name="faithfulness_gpt3", llm=gpt3, batch_size=3
)
subset = df.iloc[:3]
hg_dataset_1 = Dataset(pa.Table.from_pandas(subset))
result = evaluate(hg_dataset_1, metrics=[faithfulness_gpt3, context_relevancy, answer_relevancy])```

**Comment by shahules786** [comment]
Thanks for reporting this issue @PseudoKush . This is very intriguing, we also haven't observed such results before. I know this would be difficult but if there is any way for you to share the row data, I can try and reproduce the issue.

**Comment by PseudoKush** [comment]
I have used my own dataset that was created on a list of Q&A I did within my LLM pipeline. I am not sure how can I share that since it is not available on platforms like Hugging face. What do you suggest to share the data for the particular row ?

Update on the issue : I re-ran the cell (without changing version or anything else) couple of times and score came back in the defined range of 0 - 1.

**Comment by shahules786** [comment]
If you wish to share the row you can do it as a JSON file. `dataset[idx].to_json("...")` 
This could have been caused due to inherent stochastic nature of LLM generations.

**Comment by PseudoKush** [comment]
[value_anamoly.csv](https://github.com/explodinggradients/ragas/files/12228392/value_anamoly.csv)

I have attached the csv format for the particular row. It contains question, answer, context. Using the below command, I transformed it into required form:
``` 
df["contexts"] = df.apply(lambda x: x['contexts'][58:].split('passage: ')[1:], axis=1)
hg_dataset_1 = Dataset(pa.Table.from_pandas(df))
```

**Comment by shahules786** [comment]
Thanks a lot, @PseudoKush . I was able to reproduce the error. We will raise a fix soon #80
